### DISCUSSION TITLE
unresolved external symbol ___std_find_trivial_1@12

### DISCUSSION THREAD

**Discussion by lucaramo** [googleapis/google-cloud-cpp#11929] [General]
Hi, I'm trying to use cpp libraries for the v1 "speech to text" api. I successfully built the libraries with vcpkg. then i created a small c++ console project with microsoft msvc 2019. at the time of link i get the following error:
google_cloud_cpp_common.lib(tracing_options.cc.obj) : error LNK2019: unresolved external symbol ___std_find_trivial_1@12 referenced in function "char const * __cdecl __std_find_trivial<char const ,char>(char const *,char const *,char)"

___std_find_trivial_1@12 what does this symbol refer to? How can I solve the problem?

**Comment by coryan** [comment]
>  then i created a small c++ console project with microsoft msvc 2019.

I assume you are talking about a "project" as the Visual Studio IDE understands them.  I have not actively used Visual Studio in decades, so I am mostly guessing.

> ___std_find_trivial_1@12 what does this symbol refer to? 

That looks to me like a implementation detail in the standard library.  

> How can I solve the problem?

It is hard to say, because we don't know how you compiled / linked your application.  The first thing that comes to mind is checking that your application uses the same CRT options as vcpkg used when compiling the libraries:

https://learn.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features?view=msvc-170

This discussion also seems like it could be relevant:

https://github.com/microsoft/vcpkg/issues/10289

**Comment by HypocriticalLemur** [comment]
Hey, [lucaramo](https://github.com/lucaramo), I've just had the same problem and I found the solution for me.
The problem is gone after i switched platform toolset to Visual Studio 2019 (v142)

Hope it helps
